lib/DefaultResolver.php

```diff
@@ -214,6 +218,8 @@ private function doResolve($name, array $types, $options) {
             $this->config = (yield \Amp\resolve($this->loadResolvConf()));
         }
 
+        $config = $this->loadEnvConfig();
```

Comment:
@bwoebi `putenv` is the right thing to use, not `$_ENV`. `putenv` affects calls like `system`, changing `$_ENV` does not. See `php -r 'putenv("FOO=bar"); system("env");' | grep FOO`.